src/main/java/hudson/plugins/sshslaves/SSHLauncher.java

```diff
@@ -1241,31 +1242,52 @@ public synchronized void afterDisconnect(SlaveComputer slaveComputer, TaskListen
 
             Slave n = slaveComputer.getNode();
             if (n != null && !connectionLost) {
-                // this would fail if the connection is already lost, so we want to check that.
-                // TODO: Connection class should expose whether it is still connected or not.
                 String workingDirectory = getWorkingDirectory(n);
-                String fileName = workingDirectory + "/slave.jar";
+                final String fileName = workingDirectory + "/slave.jar";
+                Future<?> tidyUp = Computer.threadPoolForRemoting.submit(new Runnable() {
+                    public void run() {
+                        // this would fail if the connection is already lost, so we want to check that.
+                        // TODO: Connection class should expose whether it is still connected or not.
 
-                SFTPv3Client sftpClient = null;
-                try {
-                    sftpClient = new SFTPv3Client(connection);
-                    sftpClient.rm(fileName);
-                } catch (Exception e) {
-                    if (sftpClient == null) {// system without SFTP
+                        SFTPv3Client sftpClient = null;
                         try {
-                            connection.exec("rm " + fileName, listener.getLogger());
-                        } catch (Error error) {
-                            throw error;
-                        } catch (Throwable x) {
-                            x.printStackTrace(listener.error(Messages.SSHLauncher_ErrorDeletingFile(getTimestamp())));
-                            // We ignore other Exception types
+                            sftpClient = new SFTPv3Client(connection);
+                            sftpClient.rm(fileName);
+                        } catch (Exception e) {
+                            if (sftpClient == null) {// system without SFTP
+                                try {
+                                    connection.exec("rm " + fileName, listener.getLogger());
+                                } catch (Error error) {
+                                    throw error;
+                                } catch (Throwable x) {
+                                    x.printStackTrace(listener.error(Messages.SSHLauncher_ErrorDeletingFile(getTimestamp())));
+                                    // We ignore other Exception types
+                                }
+                            } else {
+                                e.printStackTrace(listener.error(Messages.SSHLauncher_ErrorDeletingFile(getTimestamp())));
```

Comment:
nope it will go to `listener.error(Messages.SSHLauncher_ErrorDeletingFile(getTimestamp()))` which is a handy `PrintStream` sending to the listener